source/PlayerInfo.cpp

```diff
@@ -1346,6 +1346,17 @@ void PlayerInfo::ReorderShip(int fromIndex, int toIndex)
 	shared_ptr<Ship> ship = ships[fromIndex];
 	ships.erase(ships.begin() + fromIndex);
 	ships.insert(ships.begin() + toIndex, ship);
+	// Check if the ship in the first position can be a flagship and is in the current system.
+	auto firstShipIt = ships.begin();
+	if(Preferences::Has("Automatically unpark flagship")
+			&& (*firstShipIt)->CanBeFlagship() && (*firstShipIt)->GetSystem() == system)
+	{
+		if(fromIndex == 0)
+			ships[toIndex]->SetIsParked(true);
+		else
+			ships[fromIndex]->SetIsParked(true);
+		(*firstShipIt)->SetIsParked(false);
```

Comment:
This will park the former flagship even if the ship you switched it with was unparked. I don't think it's necessary to park the flagship at all.

Reply:
I think the behavior you are asking for was in the first state of the pr. It was later asked to change the previous flagship to parked automatically.